ipn/ipnlocal/local.go

```diff
@@ -801,6 +804,43 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 	}
 
 	b.mu.Lock()
+
+	// Handle node expiry in the netmap, or zero if none.
+	var nextExpiry time.Time
+	if st.NetMap != nil {
+		b.em.flagExpiredPeers(st.NetMap)
+
+		// Figure out when the next node in the netmap is expiring so we can
+		// start a timer to reconfigure at that point.
+		for _, peer := range st.NetMap.Peers {
+			if peer.KeyExpiry.IsZero() {
+				continue // tagged node
+			}
+			if peer.KeyExpiry.Before(nextExpiry) {
```

Comment:
This will never be true. Nothing is before the zero value.

You need nextExpiry.IsZero() || this.